src/main/java/no/nav/familie/integrasjoner/oppgave/OppgaveService.kt

```diff
@@ -72,61 +74,89 @@ class OppgaveService constructor(
         }
+
+        val nySaksbehandlerIdent = saksbehandlerService.hentNavIdent(saksbehandler)
+
+        val oppdatertOppgaveDto = oppgave.copy(
+            id = oppgave.id,
+            versjon = versjon ?: oppgave.versjon,
+            tilordnetRessurs = nySaksbehandlerIdent,
+            beskrivelse = lagOppgaveBeskrivelseFordeling(oppgave = oppgave, nySaksbehandlerIdent = nySaksbehandlerIdent),
+        )
+        oppgaveRestClient.oppdaterOppgave(oppdatertOppgaveDto)
+
         return oppgaveId
     }
 
     fun tilbakestillFordelingPåOppgave(oppgaveId: Long, versjon: Int?): Long {
-        if (versjon != null) {
-            oppgaveRestClient.oppdaterOppgave(Oppgave(id = oppgaveId, versjon = versjon, tilordnetRessurs = ""))
-        } else {
-            val oppgave = oppgaveRestClient.finnOppgaveMedId(oppgaveId)
-
-            if (oppgave.status === StatusEnum.FERDIGSTILT) {
-                throw OppslagException(
-                    "Kan ikke tilbakestille fordeling på oppgave med id $oppgaveId som allerede er ferdigstilt",
-                    "Oppgave.tilbakestill",
-                    Level.LAV,
-                    HttpStatus.BAD_REQUEST,
-                )
-            }
+        val oppgave = oppgaveRestClient.finnOppgaveMedId(oppgaveId)
 
-            val oppdatertOppgaveDto = oppgave.copy(
-                id = oppgave.id,
-                versjon = oppgave.versjon,
-                tilordnetRessurs = "",
+        if (oppgave.status === StatusEnum.FERDIGSTILT) {
+            throw OppslagException(
+                "Kan ikke tilbakestille fordeling på oppgave med id $oppgaveId som allerede er ferdigstilt",
+                "Oppgave.tilbakestill",
+                Level.LAV,
+                HttpStatus.BAD_REQUEST,
             )
-            oppgaveRestClient.oppdaterOppgave(oppdatertOppgaveDto)
         }
 
+        if (versjon != null && versjon != oppgave.versjon) {
+            throw OppslagException(
+                "Kan ikke fordele oppgave med id $oppgaveId fordi det finnes en nyere versjon av oppgaven.",
+                "Oppgave.fordel",
+                Level.LAV,
+                HttpStatus.CONFLICT,
+            )
+        }
+
+        val oppdatertOppgaveDto = oppgave.copy(
+            id = oppgave.id,
+            versjon = versjon ?: oppgave.versjon,
+            tilordnetRessurs = "",
+            beskrivelse = lagOppgaveBeskrivelseFordeling(oppgave = oppgave),
+        )
+        oppgaveRestClient.oppdaterOppgave(oppdatertOppgaveDto)
+
         return oppgaveId
     }
 
+    private fun lagOppgaveBeskrivelseFordeling(oppgave: Oppgave, nySaksbehandlerIdent: String? = null): String {
+        val innloggetSaksbehandlerIdent = SikkerhetsContext.hentSaksbehandler()
+        val saksbehandlerNavn = SikkerhetsContext.hentSaksbehandlerNavn(strict = true)
+
+        val formatertDato = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy.MM.dd' 'HH:mm"))
```

Comment:
`date-patterns` kan være fint å ta ut i konstanter
Ikke for at det har så mye å si for oss, men koster litt å generere den på nytt hver gang hvis jeg husker det riktig

Reply:
Ah, det visste jeg ikke :) Takk for innspill, fikset nå